### ISSUE TITLE
Game Initially Renders at Full Resolution Until Window Resize

### ISSUE THREAD

**Issue by pengolod** [juliand665/retiNO#2]
<img width="1392" alt="Screen Shot 2021-04-29 at 8 39 05 PM" src="https://user-images.githubusercontent.com/80605795/116638006-f4759800-a92a-11eb-9e67-aab885c5cb9f.png">

For some reason when the window first renders, it renders at full resolution, with the correctly sized window... once I resize the window it fixes itself.

ty for making this mod regardless!!! It's wonderful to finally get the fps increases that Sodium purports! I get 110fps now :D

**Comment by pengolod** [comment]
Maybe there's a function that could be called to "resize" the window to the same size it started as....

**Comment by juliand665** [comment]
Oof yeah, I think I remember having issues like that when I needed the mod myself. The problem is that the mod is incredibly simple right now and pretty much universally compatible with any version of Minecraft, and adding some workaround to this issue would necessarily make it more complex, less robust, and prone to needing updates in the future.

Personally I just immediately take the game into fullscreen with the green button anyway, which also fixes it (since it's essentially a resize).

**Comment by pengolod** [comment]
That makes sense - I was hoping for a solution since I start the game with a 16:9 (+ 30px for the menu) window resolution so that I can record it. I'm curious if there's a similarly universal fix for this...

**Comment by juliand665** [comment]
For what it's worth, taking it into macOS fullscreen and back preserves the size you had set, it just takes a couple seconds to go through its animations.

Why not record in fullscreen, anyway?

**Comment by pengolod** [comment]
Good point, that's a useful stopgap anyways.. I've got a 16:10 display, since I'm on a Macbook Pro. I wonder now if there's also an option to force Minecraft to render at 16:9 lmao

**Comment by juliand665** [comment]
Ah, I see. Personally, I'd probably just record and upload in 16:10 anyway, I don't think people will mind the letterboxing (or rather pillarboxing) too much. If you have the money, an external display in 1080p might also be an option—shouldn't be too too expensive, anyway.

**Comment by pengolod** [comment]
The other problem is that full-screen can't be tabbed out of, so I have to drag it to a separate virtual desktop, so I usually find windowed mode easier to use. Though I guess clicking the green button fullscreen works too

**Comment by juliand665** [comment]
Yeah, I always use the macOS fullscreen, i.e. the green button. I set up a keybind in [BetterTouchTool](https://folivora.ai) (can't recommend that program enough) to take my current window fullscreen with cmd-alt-ctrl-enter, which works even when the app itself doesn't implement the standard macOS shortcut of cmd-ctrl-F.

**Comment by pengolod** [comment]
<img width="997" alt="image" src="https://user-images.githubusercontent.com/80605795/116953112-0c625a00-ac52-11eb-8490-4676caa6fd99.png">

I guess I'll try that out, thx!

**Comment by pengolod** [comment]
I've got Discord setup with ctrl alt cmd M, D, and V for toggling mute, deafen and voice activity - seems like a useful key combo

**Comment by intergrav** [comment]
Is this still an issue? Considering adding this to [Adrenaline](https://modrinth.com/modpack/adrenaline) and [Additive](https://modrinth.com/modpack/additive) modpacks. Don't have a mac to test with sadly

**Comment by juliand665** [comment]
Unfortunately yes, I just checked. That is kind of a dealbreaker for anyone not installing the mod themselves and deciding to accept it…

If you're looking for other ways to lighten the load, [Dynamic FPS](https://modrinth.com/mod/dynamic-fps) (also by me lol) helps by massively reducing the GPU usage of Minecraft when it's in the background, which on my old MacBook Pro made the difference between the machine struggling to do anything else and feeling like MC wasn't even open.
